File: answers/leetcode/145. Binary Tree Postorder Traversal/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.dfs(root, result)
        return result

    def dfs(self, root: Optional[TreeNode], result: List[int]):
        if root == None:
            return

        self.dfs(root.left, result)
        self.dfs(root.right, result)
        result.append(root.val)

File: answers/leetcode/145. Binary Tree Postorder Traversal/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> traversal = new ArrayList<>();
        if (root != null) {
            traversal.addAll(this.postorderTraversal(root.left));
            traversal.addAll(this.postorderTraversal(root.right));
            traversal.add(root.val);
        }
        return traversal;
    }
}